c-bindings/src/data_serializable_raw.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iterator>
+
+// C++ client public headers
+#include "geode/DataInput.hpp"
+#include "geode/DataOutput.hpp"
+#include "geode/internal/DataSerializablePrimitive.hpp"
+
+// C client public headers
+#include "data_serializable_raw.hpp"
+
+// C client private headers
+
+using apache::geode::client::DataInput;
+using apache::geode::client::DataOutput;
+
+namespace apache {
+namespace geode {
+namespace client {
+namespace internal {
+
+DataSerializableRaw::DataSerializableRaw(const int8_t* data, size_t size) {
```

Comment:
Ideally, the C# layer would have access to the DataInput and DataOutput buffers via the Region. However, this would likely require moving too much buffer management code to C#. An alternative that might be worth consideration is to have an interop buffer in the C++ heap (an std::vector), and give C# its address. This could be created during the cache create and deleted when the cache is closed. Always passing the length(s) of interop parameters would allow growing the buffer as needed.

Reply:
To an extent this is a "six of one and half a dozen of another" issue, but I suppose it is a little bit cleaner to just go with one parameter of type `std::vector` in the ctor, then use `std:move` move to copy it into bytes_.